### ISSUE TITLE
Support mongoose 6.13.5 for typegoose 10.5.x to fix CVE-2024-53900

### ISSUE THREAD

**Issue by sschmeck** [typegoose/typegoose#971]
## Which feature

Currently we are using mongoose in version 6.12.9 with typegoose 10.5.0
In the mentioned mongoose version the vulnerability [CVE-2024-53900](https://nvd.nist.gov/vuln/detail/CVE-2024-53900) exists.
 
Now we would like to update to the patched version of mongoose 6.13.5.
When doing so, the `npm ci` command fails:
 
```
$ npm ci
npm error code ERESOLVE
npm error ERESOLVE could not resolve
npm error
npm error While resolving: @typegoose/typegoose@10.5.0
npm error Found: mongoose@6.13.5
npm error node_modules/mongoose
npm error   mongoose@"6.13.5" from the root project
npm error
npm error Could not resolve dependency:
npm error peer mongoose@"~6.12.0" from @typegoose/typegoose@10.5.0
npm error node_modules/@typegoose/typegoose
npm error   @typegoose/typegoose@"10.5.0" from the root project
npm error
npm error Conflicting peer dependency: mongoose@6.12.9
npm error node_modules/mongoose
npm error   peer mongoose@"~6.12.0" from @typegoose/typegoose@10.5.0
npm error   node_modules/@typegoose/typegoose
npm error     @typegoose/typegoose@"10.5.0" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution.
```
 
Checking [typegoose compatibility matrix](https://typegoose.github.io/typegoose/versions/10.x/docs/guides/mongoose-compatibility), typegoose 10.5.x only supports ~6.12.0
 
When using "--legacy-peer-deps" the "nom ci" command works.
But are these version compatible or is there some work on it planned?

**Comment by smolinari** [comment]
@sschmeck 

Hey Sebastian,

Your link goes to the old docs. This is the up-to-date compatibility chart. 

https://typegoose.github.io/typegoose/docs/guides/mongoose-compatibility

Scott

**Comment by hasezoey** [comment]
I was not aware there was a new minor mongoose version for that major. I try to support all mongoose versions that are not EOL in typegoose, though only officially the latest are supported.

**Comment by hasezoey** [comment]
This has been fixed with 5b1a75f327184bd222d40eaac2fbe123f36e8967, released as [10.6.0](https://github.com/typegoose/typegoose/releases/tag/v10.6.0)

**Comment by sschmeck** [comment]
@hasezoey, thank you very much for your efforts. It saves us the migration to the newer major version for now (since we have a tight timeline for our product).
